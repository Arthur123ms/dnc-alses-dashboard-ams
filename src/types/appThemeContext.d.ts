export interface AppThemeConstextProps {
    appTheme: string
    toggleTheme: () => void
}