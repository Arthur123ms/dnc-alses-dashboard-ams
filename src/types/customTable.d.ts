export interface CustomTableProps {
    header: string[]
    rows: React.ReactNode[][]
}