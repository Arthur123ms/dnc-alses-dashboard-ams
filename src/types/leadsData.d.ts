export interface LeadsPostData{
    name: string
    email: srting
    phone: string
}

export interface LeadsData extends LeadsPostData{
    id: number
}
