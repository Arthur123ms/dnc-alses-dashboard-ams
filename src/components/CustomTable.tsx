import styled from 'styled-components'
import { type CustomTableProps } from '@/types'
import { pxToRem } from '@/utils'

const TableWrapper = styled.div`
    overflow-x: auto;
    widht: 100% ;
    table {
        .ellipsis{
            display: block;
            white-space: nowrap;
            text-overflow: ellipsis;
            overflow: hidden;
        }

        .ellipsis-sm{
            width: ${pxToRem(300)}
        }

        .ellipsis-xs{
            width: ${pxToRem(150)}
        }

        width: 100%;
        border-collapse: collapse;

        th,
        td {
            height: ${pxToRem(48)};
            padding: 0 ${pxToRem(8)} 0 0;
            text-align: left;
            &:last-child {
                text-align: right;
                padding: 0;
            }
        }
        
        th {
            color:  ${(props) => props.theme.typographies.subtitle};
            font-weight: 600;

        }

        tr {
            border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
            &:last-child {
                border-bottom: none;
            }
        }
    }
`
function CustomTable(props: CustomTableProps) {
    const { header, rows} = props

    return(
        
        <TableWrapper>
            <table>
                <thead>
                    <tr>
                        {
                            header.map((header, index) => (
                                <th key={index}>{header}</th>
                            ))
                        }
                    </tr>
                </thead>
                <tbody>
                    {
                        rows.map((row,rowIndex) => (
                            <tr key={rowIndex}>
                                {row.map((cell, cellIndex) => (
                                    <td key={cellIndex}>{cell}</td>
                                ))}
                            </tr>
                        ))
                    }
                </tbody>
            </table>
        </TableWrapper>
        
    
    )
}

export default CustomTable