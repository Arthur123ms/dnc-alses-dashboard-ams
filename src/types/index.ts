export * from './avatarsList.d'
export * from './customTable.d'
export * from './theme.d'
export * from './typographies.d'
export * from './formComponent.d'