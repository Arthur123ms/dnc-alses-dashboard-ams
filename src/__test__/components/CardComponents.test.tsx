import { CardComponents } from '@/components';
import {render} from '@testing-library/react';
import { type Theme } from '@/types';
import { ThemeProvider } from 'styled-components';
import { themeList } from '@/resources/themesList';

describe('Card Component', () => {
  const renderComponent = (theme: Theme, className?: string) =>
    render(
      <ThemeProvider theme={theme}>
        <CardComponents className={className} />
      </ThemeProvider>
    )

  themeList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot without any class ', () => {
        const { asFragment } = renderComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })

        it('should match the snapshot without alert class ', () => {
            const { asFragment } = renderComponent(theme, 'alert')
            expect(asFragment()).toMatchSnapshot()
        })

        it('should match the snapshot without sucess class ', () => {
        const { asFragment } = renderComponent(theme, 'sucess')
        expect(asFragment()).toMatchSnapshot()
      })

        it('should match the snapshot without warning class ', () => {
        const { asFragment } = renderComponent(theme, 'warning')
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})