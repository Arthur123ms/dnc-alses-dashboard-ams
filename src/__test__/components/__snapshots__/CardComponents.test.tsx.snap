// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Card Component Dark Theme should match the snapshot without alert class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`Card Component Dark Theme should match the snapshot without any class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Card Component Dark Theme should match the snapshot without sucess class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 sucess"
  />
</DocumentFragment>
`;

exports[`Card Component Dark Theme should match the snapshot without warning class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`Card Component Light Theme should match the snapshot without alert class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`Card Component Light Theme should match the snapshot without any class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Card Component Light Theme should match the snapshot without sucess class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 sucess"
  />
</DocumentFragment>
`;

exports[`Card Component Light Theme should match the snapshot without warning class  1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.sucess {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;
